lib/reports/ReportPeakHourFactor.js

```diff
@@ -0,0 +1,323 @@
+/* eslint-disable class-methods-use-this */
+import ArrayUtils from '@/lib/ArrayUtils';
+import { ReportBlock, ReportType } from '@/lib/Constants';
+import ReportBaseFlow from '@/lib/reports/ReportBaseFlow';
+import ReportBaseFlowDirectional from '@/lib/reports/ReportBaseFlowDirectional';
+import ReportCountSummaryTurningMovement from '@/lib/reports/ReportCountSummaryTurningMovement';
+import {
+  indexRangeAm,
+  indexRangeMax,
+  indexRangePeakHour,
+  indexRangePm,
+  indexRangeSum,
+} from '@/lib/reports/time/ReportTimeUtils';
+import TimeFormatters from '@/lib/time/TimeFormatters';
+
+const dirs = ['N', 'E', 'S', 'W'];
+const turns = ['L', 'T', 'R'];
+
+class ReportPeakHourFactor extends ReportBaseFlow {
+  type() {
+    return ReportType.PEAK_HOUR_FACTOR;
+  }
+
+  static getEmptyPeakData(countData, lo) {
+    let start = countData[lo].t;
+    start = start.minus({
+      minutes: ReportBaseFlow.MINUTES_PER_ROW,
+    });
+    const peakData = {
+      timeRange: { start, end: start },
+      movement: {},
+      direction: {},
+      total: null,
+    };
+    dirs.forEach((dir) => {
+      peakData.direction[dir] = null;
+      turns.forEach((turn) => {
+        peakData.movement[`${dir}_${turn}`] = null;
+      });
+    });
+    return peakData;
+  }
+
+  static getPeakHourFactor(countData, indexRangePeak, fnVolume) {
+    const volumeTotal = indexRangeSum(countData, indexRangePeak, fnVolume);
+    const volumeMax = indexRangeMax(countData, indexRangePeak, fnVolume);
+    if (volumeMax === 0) {
+      return null;
+    }
+    return volumeTotal / (4 * volumeMax);
```

Comment:
this still assumes that `countData` contains 15-minute buckets, but we can revisit that assumption more fully later.  (please document it for now, though, so that we have an easier time changing this later if need be.)